src/twin/provisioning_config.rs

```diff
@@ -164,20 +176,20 @@ impl ProvisioningConfig {
     there should not be an issue here. thus we suppress clippy warning.
     */
     #[allow(clippy::await_holding_refcell_ref)]
-    pub async fn next(&self) -> impl future::Future<Output = ()> {
+    pub async fn next(&self) {
         let Some(ref rx) = self.rx_identity_watcher else {
             debug!("provisioning_config watcher: not active");
-            return future::pending().right_future();
+            return future::pending().await;
         };
 
         let Some(_) = rx.borrow_mut().recv().await else {
             debug!("provisioning_config watcher: sender dropped");
-            return future::pending().right_future();
+            return future::pending().await;
         };
 
         info!("provisioning_config watcher: config changed");
 
-        future::ready(()).left_future()
+        future::ready(()).await
```

Comment:
Not quite sure whether I understand what's happening here. Is future::pending().await really what we want? According to the docs this future will never resolve? https://doc.rust-lang.org/std/future/fn.pending.html Doesn't this mean, that the function will never return in the error cases? Conversely, won't future::ready(()) not return immediately? Do we need it then, after all? Wouldn't it be sufficient to use return only in any cases?

Reply:
This is not the head of branch you reviewed here. The code you commented is not existing anymore.

Reply:
Ack, my bad :)